### PR TITLE
Allow filename option to be an absolute path.

### PR DESCRIPTION
**Pull request by amyboyd**

### CODE REVIEW

**Comment by sindresorhus** [comment]
Why?

**Comment by amyboyd** [comment]
Why not? I have a use-case where jobs have different current-working-directories depending on what they are doing. Putting the files into /tmp or another place instead of always into the current-working-directory makes sense for this case, and I can think of other cases.

**Comment by sindresorhus** [comment]
> Why not?

Simplicity. If you ask me that the answer is always no.

-

I assume you're talking about the CLI not API.

So why not just change the cwd before running Pageres?

**Comment by amyboyd** [comment]
This way is so much simpler than changing a PHP/Java/Go/whatever's CWD. Changing the CWD has knock-on effects on the rest of the code. This seems like an obvious simplification.

**Comment by sindresorhus** [comment]
Hmm, then what you really looking for is a `--dest` option. That way it would support multiple files too.

**Comment by amyboyd** [comment]
I disagree. If you don't want to merge, that's fine, but I can't see how this isn't an obvious win.

**Comment by sindresorhus** [comment]
Because it suddenly makes the `filename` option be a path too. Bear with me. I'm trying to find an optimal solution for both of us.

How would this not work for you `pageres website.com 1000x1000 --dest='/tmp'`?

**Comment by amyboyd** [comment]
It's not standard for any tool to have a different directory and filename option. `ls`? `cp`? `fs.readFileSync`? All take full paths.

**Comment by sindresorhus** [comment]
Yes, but those tools don't support multiple files at once.

You can see the same behavior in gulp where you specify a destination folder: https://github.com/gulpjs/gulp/blob/master/docs/API.md#gulpdestpath-options

@kevva Thoughts?

File: lib/util.js [hunk 1]
```diff
@@ -145,7 +147,11 @@ exports.save = function (streams, cb) {
 exports.create = function (uri, size, options) {
 	var sizes = size.split('x');
 	var stream = screenshot(protocolify(uri), size, options);
-	var filename = _.template(options.filename + '.png', {
+
+	var filenameWithPngExtension = (options.filename.substr(-4) === '.png'
```
Comment (kevva) [inline lib/util.js]: Since we support multiple formats this is not reliable.

**Comment by kevva** [comment]
> I assume you're talking about the CLI not API.

@amyboyd? If we're talking about the API I'm not totally against adding a `dest` prop to options that superseds the `.dest()` method. And a `--dest` flag to the CLI, but not changing `--filename`.

**Pull request closed without merging**